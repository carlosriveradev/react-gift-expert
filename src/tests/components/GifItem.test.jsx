/**
* @jest-environment jsdom
*/
import { render, screen } from '@testing-library/react'
import { GifItem } from "../../components/GifItem";

describe('Pruebas en el <GifItem />', () => {

    const title = 'goku'
    const url = 'https://goku.com/gohan'

    test('debe de hacer match con el snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect( container ).toMatchSnapshot()
    })
    test('debe mostrar la imagen con el URL y el ATL indicado', () => {
        render(<GifItem title={title} url={url} />)
        // screen.debug()
        // expect(screen.getByRole('img').src).toBe(url)
        // expect(screen.getByRole('img').alt).toBe(title)
        const { src, alt } = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)
    })
    test('debe mostrar el titulo en el componente', () => {
        render(<GifItem title={title} url={url} />)
        expect(screen.getByText(title)).toBeTruthy()
    })
})