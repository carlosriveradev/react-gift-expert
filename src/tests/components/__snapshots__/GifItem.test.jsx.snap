// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifItem /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="goku"
      src="https://goku.com/gohan"
    />
    <p>
      goku
    </p>
  </div>
</div>
`;
